Services/TokenAuthentication/ModsenOnlineStore.Login.Infrastructure/Services/LoginService.cs

```diff
@@ -50,48 +54,99 @@ public async Task<DataResponseInfo<string>> GetToken(LoginData data)
             );
 
-            return new DataResponseInfo<string>(new JwtSecurityTokenHandler().WriteToken(jwt), true, "token");
+            return new DataResponseInfo<string>(data: new JwtSecurityTokenHandler().WriteToken(jwt), success: true, message: "token");
         }
 
         public async Task<DataResponseInfo<List<User>>> GetAllUsers() =>
-            new DataResponseInfo<List<User>>(await repository.GetAllUsers(), true, "all users");
+            new DataResponseInfo<List<User>>(data: await repository.GetAllUsers(), success: true, message: "all users");
 
         public async Task<DataResponseInfo<User>> GetUserById(int id)
         {
             var user = await repository.GetUserById(id);
-            if (user is null) return new DataResponseInfo<User>(null, true, "user not found");
+            if (user is null) return new DataResponseInfo<User>(data: null, success: false, message: "user not found");
 
-            return new DataResponseInfo<User>(user, true, $"user with id {user.Id}");
+            return new DataResponseInfo<User>(data: user, success: true, message: $"user with id {user.Id}");
         }
 
-
-        public async Task<DataResponseInfo<List<User>>> RegisterUser(AddUserDto userDto)
+        public async Task<ResponseInfo> RegisterUser(AddUserDto userDto)
         {
-            if (userDto is null) return new DataResponseInfo<List<User>>(null, false, "wrong request data");
+            if (userDto is null) return new ResponseInfo(success: false, message: "wrong request data");
             var newUser = mapper.Map<User>(userDto);
 
-            var users = await repository.RegisterUser(newUser);
+            await repository.RegisterUser(newUser);
+
+            var user = await repository.GetUserByEmail(newUser.Email);
+
+            var emailConfirmation = new EmailConfirmation
+            {
+                UserId = user!.Id,
+                Code = Guid.NewGuid().ToString()
+            };
+
+            await emailConfirmationRepository.AddEmailConfirmation(emailConfirmation);
+
+            var factory = new ConnectionFactory() { HostName = "localhost" };
+            using (var connection = factory.CreateConnection())
+            using (var channel = connection.CreateModel())
+            {
+                channel.QueueDeclare(queue: "email-confirmation",
+                                     durable: false,
+                                     exclusive: false,
+                                     autoDelete: false,
+                                     arguments: null);
+
+                var message = user.Email + " " + user.Id + " " + emailConfirmation.Code;
+
+                var body = Encoding.UTF8.GetBytes(message);
+
+                channel.BasicPublish(exchange: "",
+                                     routingKey: "email-confirmation",
+                                     basicProperties: null,
+                                     body: body);
```

Comment:
move all stuff about messaging to another abstraction

Reply:
fixed